src/zope/configuration/tests/test_config.py

```diff
@@ -1465,6 +1498,12 @@ def _handler(context, **kw):
 
 class _ConformsToIComplexDirectiveContext(object):
 
+    def _getTargetClass(self):
+        raise NotImplementedError
+
+    def _makeOne(self, *args, **kwargs):
+        raise NotImplementedError
```

Comment:
I would be mildly happier if we could distinguish `raise NotImplementedError("please override in subclasses")` from `raise NotImplementedError("not called by current tests")`, but it's probably a lot of work for virtually no gain.

Just thinking out loud here, feel free to ignore.

Reply:
In the past, I've made that distinction by raising `AssertionError` from the "should not be called" cases. I don't know why I didn't do that here. It's probably a pretty easy cleanup to do.